src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/maven/WriteMavenPomStep.java

```diff
@@ -116,6 +127,19 @@ public String getFunctionName() {
         public String getDisplayName() {
             return "Write a maven project file.";
         }
+
+        @Nonnull
+        @Override
+        public Map<String, Object> filterForAction(@Nonnull Step step, @Nonnull Map<String, Object> map) {
+            if (step instanceof WriteMavenPomStep) {
+                Map<String, Object> args = new HashMap<>(map);
+                if (args.containsKey("model")) {
+                    args.replace("model", ArgumentsAction.NotStoredReason.MASKED_VALUE);
```

Comment:
Why does it need to be masked? Because it cannot be XStream serialized? Then that should be handled in `workflow-api` automatically.

Reply:
Does https://github.com/jenkinsci/workflow-cps-plugin/pull/220 help?